### PR TITLE
Setup backend

### PR DESCRIPTION
**Pull request by macben888**

### CODE REVIEW

**Review by Laifson — APPROVED** [review]

File: backend/src/main/java/capstone/backend/controller/EmployeeController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@RequestMapping("api/employee")
+public class EmployeeController {
+
+    private final EmployeeService service;
+
+    public EmployeeController(EmployeeService service) {
+        this.service = service;
+    }
+
+    @GetMapping("")
```
Comment (Cobas91) [inline backend/src/main/java/capstone/backend/controller/EmployeeController.java:21]:
```suggestion
    @GetMapping
```
Dont need a empty string here.

File: backend/src/main/java/capstone/backend/security/service/UserAuthUtils.java [hunk 2]
```diff
@@ -5,12 +5,8 @@
 
 @Component
 public class UserAuthUtils {
-    //TODO implement better password validation
     public void validatePassword(String password) throws InvalidCredentialsException {
         if(password.length() < 3) throw new InvalidCredentialsException("Password too short");
```
Comment (Cobas91) [inline backend/src/main/java/capstone/backend/security/service/UserAuthUtils.java:9]: Better do the password validation in frontend, dont use ressources from backend when you could validate the data before.

File: backend/src/test/java/capstone/backend/controller/EmployeeControllerTest.java [hunk 3]
```diff
@@ -1,7 +1,78 @@
...
+    @Autowired
+    PasswordEncoder passwordEncoder;
+    @Autowired
+    EmployeeMapper mapper;
+
+    @Container
+    public static PostgreSQLContainer container = new PostgreSQLContainer()
+            .withDatabaseName("quiz")
+            .withUsername("quiz")
+            .withPassword("quiz");
```
Comment (Cobas91) [inline backend/src/test/java/capstone/backend/controller/EmployeeControllerTest.java:51]: Copy paste issue from other project

File: backend/src/test/java/capstone/backend/security/controller/AuthControllerTest.java [hunk 4]
```diff
@@ -84,71 +72,48 @@ void login() {
...
     @Test
     void loginFailsWithWrongUsername() {
-        UserDTO user = sampleUser();
+        EmployeeDTO user = TestUtils.sampleUserDTO();
         user.setPassword("1234");
         user.setUsername("wrong_username");
-        userRepository.save(mapper.mapUser(user));
+        repo.save(EmployeeMapper.mapEmployee(user));
         user.setPassword("123");
         ResponseEntity<String> response = restTemplate.postForEntity("/auth/login", user, String.class);
```
Comment (Cobas91) [inline backend/src/test/java/capstone/backend/security/controller/AuthControllerTest.java:89]: You changed the login path to "api/employee" in the scurity config?

**Review by Cobas91 — APPROVED** [review]
next time splitt backend and frontend into two branches. Also dont create empty files. Create them when you need them. much scss files are not needed yet.

**Pull request merged**